viewing_party/party.py

```diff
@@ -0,0 +1,287 @@
+
+    # Create a dictionary where the keys are
+    # title, genre, and rating of
+    # the new dictionary - grab these values
+    # from the arguments passed as parameters
+
+    # if every argument that is passed is truthy
+    # then run the code below, otherwise:
+    # run different code
+
+    if movie_title and genre and rating:
+        movie = {}
+        movie["title"] = movie_title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    """
+    input: A dictionary of user_data and a particular movie
+    output: Returns the dictionary of user_data, with the movie added to the "watched" key
+    """
+    # Given a movie dictionary, 
+    # Return statement for this function should equal the dictionary of user_data
+    # user_data is a parameter of data type dictionary
+    # movie is also a parameter of data type dictionary
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """
+    input: A dictionary of user_data and a particular movie
+    output: Returns the user_data dictionary with the movie added to the "watchlist" key of user_data
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    """
+    input: A dictionary of user_data and a movie title
+    output: The user_data dictionary with the movie dictionary removed from "watchlist"
+    and added to the "watched" list
+    """
+    # Check whether movie_title was in watchlist
+    # user_data[watchlist]
+    # If it was, remove it from the watchlist
+    # Append it to the add_to_watched list
+
+    # Watchlist is a list with multiple dictionaries
+    # Loop through each element in the watchlist list
+    # Check whether the title key matches the movie_title given above
+
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            watched_movie = movie
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(watched_movie)
+
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    """
+    input: A dictionary of user_data
+    output: The average rating of each movie in the user's watched list
+    """
+
+    # Access the watched key in user_data
+    # That points to a list
+    # Inside this list are dictionary elements
+    # We need to loop through each dictionary element
+    # in watched, access the rating keys
+    # store them in a variable for calculation
+    # calculate the average of each
+
+    sum = 0
+    avg_list = []
+
+    # store each rating int into a new list
+
+    if not user_data["watched"]:
+        return 0.0
+
+    else:
+        for movie in user_data["watched"]:
+            avg_list.append(movie["rating"])
+
+        for num in avg_list:
+            sum += num
+        
+        avg = sum / len(avg_list)
+        
+        return avg
```

Comment:
The `else` here is not necessary. 
```suggestion

    for movie in user_data["watched"]:
          avg_list.append(movie["rating"])

    for num in avg_list:
          sum += num
        
    avg = sum / len(avg_list)
        
    return avg
```